openedx/features/lti_course_tab/tab.py

```diff
@@ -0,0 +1,245 @@
+from typing import Dict
+from urllib.parse import quote
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib.sites.shortcuts import get_current_site
+from django.http import HttpRequest
+from django.utils.translation import get_language, to_locale, ugettext_lazy
+from lti_consumer.lti_1p1.contrib.django import lti_embed
+from lti_consumer.models import LtiConfiguration
+from opaque_keys.edx.keys import CourseKey
+from web_fragments.fragment import Fragment
+
+from lms.djangoapps.courseware.access import get_user_role
+from lms.djangoapps.courseware.tabs import EnrolledTab
+from openedx.core.djangoapps.discussions.models import DiscussionsConfiguration
+from openedx.core.djangolib.markup import HTML
+from common.djangoapps.student.models import anonymous_id_for_user
+from xmodule.course_module import CourseDescriptor
+from xmodule.tabs import TabFragmentViewMixin, key_checker
+
+
+class LtiCourseLaunchMixin(TabFragmentViewMixin):
+    """
+    Mixin that encapsulates all LTI-related functionality from the View
+    """
+
+    ROLE_MAP = {
+        'student': u'Student',
+        'staff': u'Administrator',
+        'instructor': u'Instructor',
+    }
+    DEFAULT_ROLE = u'Student'
+
+    def get_additional_lti_parameters(self, course: CourseDescriptor, request: HttpRequest) -> Dict[str, str]:
+        lti_config = self.get_lti_config(course)
+        additional_config = lti_config.lti_config.get('additional_parameters', {})
+        return additional_config
+
+    @staticmethod
+    def get_user_id(user: User, course_key: CourseKey):
+        return quote(anonymous_id_for_user(user, course_key))
+
+    def get_lti_roles(self, user: User, course_key: CourseKey) -> str:
```

Comment:
Just noting the discrepancy between `get_lti_roleS` (plural) and `self.default_rolE`/`get_user_rolE` (both singular).

Should this be `get_lti_role` or is there are reason to have them be different?

Reply:
It's a base class, and technically it's possible to pass on multiple roles, as a comma-separated list. We're using a single entry but it's possible that in the future this or  another subclass of this might provide multiple roles.